### PR TITLE
refactor(cli): refactor storage manager

### PR DESCRIPTION
**Pull request by iapershin**
Added constructor for storage manager and deduplicated related code

### CODE REVIEW

File: cmd/werf/common/storage.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	return func(config *NewStorageManagerConfig) {
+		config.hostPurge = true
+	}
+}
+
+func NewStorageManager(ctx context.Context, c *NewStorageManagerConfig) (*manager.StorageManager, error) {
+	return NewStorageManagerWithOptions(ctx, c)
+}
+
+func NewStorageManagerWithOptions(ctx context.Context, c *NewStorageManagerConfig, opts ...Option) (*manager.StorageManager, error) {
```
Comment (alexey-igrychev) [inline cmd/werf/common/storage.go:31]: Optional args (`opts`) are not handled.

File: cmd/werf/common/storage.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/werf/werf/v2/pkg/container_backend"
+	"github.com/werf/werf/v2/pkg/storage/manager"
+)
+
+type Option func(*NewStorageManagerConfig)
```
Comment (alexey-igrychev) [inline cmd/werf/common/storage.go:11]: We need a separate package or unique name.

File: cmd/werf/common/storage.go (hunk 1 above)
Comment (iapershin) [reply cmd/werf/common/storage.go]: resolved

File: cmd/werf/common/storage.go (hunk 2 above)
Comment (iapershin) [reply cmd/werf/common/storage.go]: resolved

**Review by alexey-igrychev — APPROVED** [review]

**Pull request merged**